src/komodo_utils.cpp

```diff
@@ -1239,21 +1359,79 @@ int8_t equihash_params_possible(uint64_t n, uint64_t k)
     return(-1);
 }
 
+/***
+ * @brief get username, password, and port from a config file
+ * @param[in] path the path to the data directory
+ * @param[in] filename the filename of the config file (without directory)
+ * @param[out] userpass the username and password from the config file (colon separated)
+ * @param[out] port the RPC port found in the config file
+ */
+void get_userpass_and_port(const boost::filesystem::path& path, const std::string& filename, 
+        std::string& userpass, uint16_t& port)
+{
+    userpass = "";
+    port = 0;
+    boost::filesystem::path datadir_path = path;
+    datadir_path /= filename;
+    FILE* fp = fopen(datadir_path.string().c_str(), "rb");
+    if ( fp != nullptr )
+    {
+        char username[512];
+        char password[4096];
+        port = _komodo_userpass(username,password,fp);
+        if ( username[0] != 0 && password[0] != 0 )
+            userpass = std::string(username) + ":" + std::string(password);
+        fclose(fp);
+    } 
+    else 
+        printf("couldnt open.(%s) will not validate dest notarizations\n", datadir_path.string().c_str());
+}
+
+/****
+ * @brief set ports and usernames/passwords from command line and/or config files
+ * @note modifies ASSETCHAINS_P2PPORT, ASSETCHAINS_RPCPORT, KMDUSERPASS, BTCUSERPASS, DESTPORT
+ * @note IS_KOMODO_NOTARY should already be set
+ * @param ltc_config_filename configuration file for ltc (via -notary command line parameter)
+ */
+void set_kmd_user_password_port(const std::string& ltc_config_filename)
+{
+        ASSETCHAINS_P2PPORT = 7770; // default port for P2P
+        ASSETCHAINS_RPCPORT = 7771; // default port for RPC
+#ifdef __APPLE__
+        std::string filename = "Komodo.conf";
+#else
+        std::string filename = "komodo.conf";
+#endif
+
+        auto datadir_path = GetDataDir();
+        uint16_t ignore;
+        std::string userpass;
+        get_userpass_and_port(datadir_path, filename, userpass, ignore);
+        if (!userpass.empty())
+            strncpy(KMDUSERPASS, userpass.c_str(), 8705);
+        if (IS_KOMODO_NOTARY)
+        {
+            get_userpass_and_port(datadir_path, ltc_config_filename, userpass, DEST_PORT);
```

Comment:
there is an issue here: when ltc config path is created it is assumed that datadir_path points to the home dir '/home/ubuntu' to add '-notary' param .litecoin/litecoin.conf. However datadir_path is set to '/home/ubuntu/.komodo'. I suggest adding '..' to datadir_path to go one level up